ops.qmd

```diff
@@ -2,84 +2,760 @@
+
+* Operational Excellence
+    * Monitoring: Techniques for monitoring model performance, data drift, and operational health.
+    * Governance: Implementing policies for model auditability, compliance, and ethical considerations.
+
+### Model Lifecycle Management
+
+![Diagram showing the components of Data and Model Management](images/ai_ops/mlops_flow.png)
+
+#### Data Management
+
+In traditional centralized MLOps, data is aggregated into large datasets and data lakes, then processed on cloud or on-prem servers. However, embedded MLOps relies on decentralized data from local on-device sensors. Devices collect smaller batches of incremental data, often noisy and unstructured. With connectivity constraints, this data cannot always be instantly transmitted to the cloud and needs to be intelligently cached and processed at the edge.
+
+Embedded devices can only preprocess and clean data minimally before transmission due to limited on-device compute. Early filtering and processing occurs at edge gateways to reduce transmission loads. While leveraging cloud storage, more processing and storage happens at the edge to account for intermittent connectivity. Devices identify and transmit only the most critical subsets of data to the cloud.
+
+Labeling also faces challenges without centralized data access, requiring more automated techniques like federated learning where devices collaboratively label peers' data. With personal edge devices, data privacy and regulations are critical concerns. Data collection, transmission and storage must be secure and compliant.
+
+For instance, a smartwatch may collect step count, heart rate, GPS coordinates throughout the day. This data is cached locally and transmitted to an edge gateway when WiFi is available. The gateway processes and filters data before syncing relevant subsets with the cloud platform to retrain models.
+
+#### Model Training
+
+In traditional centralized MLOps, models are trained using abundant data via deep learning on high-powered cloud GPU servers. However, embedded MLOps faces severe constraints on model complexity, data availability and compute resources for training.
+
+The volume of aggregated data is much lower, often requiring techniques like federated learning across devices to create training sets. The specialized nature of edge data also limits public datasets for pre-training. With privacy concerns, data samples need to be tightly controlled and anonymized where possible.
+
+Furthermore, the models themselves need to use simplified architectures optimized for low-power edge hardware. There is no access to high-end GPUs for intensive deep learning given the compute limitations. Training leverages lower-powered edge servers and clusters with distributed approaches to spread load.
+
+![A diagram showing how transfer learning targets updates to specific layers of a model.(images/ai_ops/transfer_learning.png)
```

Comment:
Just a heads-up: this image doesn't seem to render in quarto. I think it might be because there's no end bracket after "model."

Reply:
Got fixed already by @mrdrangonbear